### PR TITLE
Unstyle empty list item on Enter key

### PR DESCRIPTION
**Pull request by ericbiewener**
**Summary**

This appears to be standard functionality in text editors. If the list item has no text and the user pressed `Enter`, rather than creating a new list item, the current one should simply be removed. This is how it works in Google Docs, TextEdit, etc.

**Test Plan**

1. Open the Rich Text editor example.
2. Create a list item. Don't add any text to it.
3. Press enter

The list item should be removed.

### CODE REVIEW

**Comment by rylev** [comment]
@ericbiewener This change would be very useful. Any chance you can sign the CLA and we can get someone from Facebook to merge this?

**Comment by ericbiewener** [comment]
@rylev I signed it a while ago. See the labels on the PR :)

**Comment by rylev** [comment]
@ericbiewener sorry about that :-D Ok I guess we just need to sit back and wait for someone from Facebook to review this.

**Comment by flarnie** [comment]
Very nice! 

As with https://github.com/facebook/draft-js/pull/762 I'd like to get a test for this, but that could be done as a follow-up. I can open an issue for the missing tests.

I'll see if we can get this merged. Great PRs! :D

**Comment by ericbiewener** [comment]
@niveditc all set. Updated tests as well.

File: src/model/transaction/splitBlockInContentState.js [hunk 1]
```diff
@@ -95,10 +96,22 @@ const splitBlockInContentState = (
...
 
   const key = selectionState.getAnchorKey();
-  const offset = selectionState.getAnchorOffset();
   const blockMap = contentState.getBlockMap();
   const blockToSplit = blockMap.get(key);
   const text = blockToSplit.getText();
+
+  if (!text) {
+    const blockType = blockToSplit.getType();
+    if (blockType && blockType.endsWith('-list-item')) {
```
Comment (niveditc) [inline src/model/transaction/splitBlockInContentState.js]: I think we should be specific about the two list item types:
`if (blockType === 'unordered-list-item' || blockType === 'ordered-list-item')`

Comment (ericbiewener) [reply src/model/transaction/splitBlockInContentState.js]: Agreed! All set.

**Comment by niveditc** [comment]
It seems that package.json still shows up in this stack & didn't get deleted on d12d823

**Comment by ericbiewener** [comment]
@niveditc Sorry, good to go for real this time :)

**Comment by niveditc** [comment]
Next steps are:
* I'll import this and run FB's test stack.
* Do a few sanity checks in the browser.
* Provided the steps above are successful I'll merge it in.

**Pull request closed without merging**